modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -254,6 +295,36 @@ typename Net::Result infer(Args&&... args) {
     return GInfer<Net>::on(std::forward<Args>(args)...);
 }
 
+struct Generic { };
+
+template<typename Net, typename... Args>
+typename std::enable_if<std::is_same<Net, Generic>::value, GInferOutputs>::type
+infer(const std::string& name, const GInferInputs& inputs)
```

Comment:
Why enable_if? Just a specialization doesn't work?

Reply:
It seems you don't need neither `Net` nor `Args...` in the below code.

Reply:
Fixed